pkg/controller/authentication/deployment.go

```diff
@@ -436,11 +441,21 @@ func generateProviderDeploymentObject(instance *operatorv1alpha1.Authentication,
 	ldapCACert := instance.Spec.AuthService.LdapsCACert
 	routerCertSecret := instance.Spec.AuthService.RouterCertSecret
 
+	metaLabels := pkgCommon.MergeMap(map[string]string{"app": deployment}, instance.Spec.Labels)
```

Comment:
instead of swapping the arguments in this function, In the MergeMap function body i have swapped it, meaning that i have considered mergeMap as a source(instance.Spec.Labels) to add back app label to mergeMap, so that we preserve app label